discord/ui/select.py

```diff
@@ -97,6 +111,82 @@
 selected_values: ContextVar[Dict[str, List[PossibleValue]]] = ContextVar('selected_values')
 
 
+def _handle_select_defaults(
+    defaults: Sequence[ValidDefaultValues],
+    component_type: Literal[
+        ComponentType.user_select,
+        ComponentType.role_select,
+        ComponentType.channel_select,
+        ComponentType.mentionable_select,
+    ],
+) -> List[SelectDefaultValue]:
+    if not defaults:
+        return []
+
+    if component_type not in (
+        ComponentType.user_select,
+        ComponentType.role_select,
+        ComponentType.channel_select,
+        ComponentType.mentionable_select,
+    ):
+        raise TypeError('component_type must be one of user_select, role_select, channel_select, or mentionable_select')
+
+    from ..app_commands import AppCommandChannel, AppCommandThread
+
+    cls_to_type: Dict[Type[ValidDefaultValues], SelectDefaultValueType] = {
+        User: SelectDefaultValueType.user,
+        Member: SelectDefaultValueType.user,
+        Role: SelectDefaultValueType.role,
+        GuildChannel: SelectDefaultValueType.channel,
+        AppCommandChannel: SelectDefaultValueType.channel,
+        AppCommandThread: SelectDefaultValueType.channel,
+        Thread: SelectDefaultValueType.channel,
+    }
+    type_to_supported_classes: Dict[ValidSelectType, Tuple[Type[ValidDefaultValues], ...]] = {
+        ComponentType.user_select: (User, Member, Object),
+        ComponentType.role_select: (Role, Object),
+        ComponentType.channel_select: (GuildChannel, AppCommandChannel, AppCommandThread, Thread, Object),
+        ComponentType.mentionable_select: (User, Member, Role, Object),
+    }
+
+    values: List[SelectDefaultValue] = []
+    for obj in defaults:
+        if isinstance(obj, SelectDefaultValue):
+            values.append(obj)
+            continue
+
+        object_type = obj.__class__ if not isinstance(obj, Object) else obj.type
+
+        print(object_type, component_type, object_type == Object)
+
+        if object_type not in type_to_supported_classes[component_type]:
+            supported_classes = [c.__name__ for c in type_to_supported_classes[component_type]]
```

Comment:
```suggestion
            # TODO: split this into a util function
            supported_classes = [c.__name__ for c in type_to_supported_classes[component_type]]
```

A note for myself please :)